float64 timestamp
float32[] spec